src/main/client/featureToggleClient.ts

```diff
@@ -0,0 +1,45 @@
+import { CoreOptions, RequestAPI } from 'request'
+import { RequestPromise } from 'request-promise-native'
+
+import { HeadersBuilder } from './headersBuilder'
+
+export class FeatureToggleClient {
+  private endpointURI: string
+  private request: RequestAPI<RequestPromise, CoreOptions, CoreOptions>
+
+  constructor (endpointURI: string,
+               request: RequestAPI<RequestPromise, CoreOptions, CoreOptions>) {
+    this.endpointURI = endpointURI
+    this.request = request
+  }
+
+  isAdmissionsAllowed (user: string, permissions: string): RequestPromise<boolean> {
```

Comment:
We should remove `isAdmissionsAllowed`, `isCitizenFrontendMaintenanceUnplannedEnabled`, `isCitizenFrontendMaintenancePlannedEnabled`, `isLegalFrontendMaintenanceUnplannedEnabled` and `isLegalFrontendMaintenancePlannedEnabled` methods and leave just:

```
isFeatureEnabled (featureName: String, user?: string, permissions?: string): RequestPromise<boolean> {
```

The reason for that is to make this module service independent so that other service teams can use it if they need. Within CMC service we would just do:

```
enum Features { // keep in some separate file
  admissions = 'cmc_admissions',
  unplannedMaintenance = 'cmc_citizen_frontend_maintenance_unplanned',
  ...
}

// upon claim save
featureToggleClient.isFeatureEnabled(Features.admissions, user.id, user.roles.join(','))

// in maintenance middleware
featureToggleClient.isFeatureEnabled(Features.unplannedMaintenance)
```

Reply:
Yes I can see you point - Code updated 👍